### PR TITLE
Improve legacy autobuyer and the new autobuyer modal

### LINKED ISSUE #2852: Legacy autobuyer issues
On the legacy VSP page, teststakepool.decred.org is configured to purchase tickets, however it cannot be selected for automatic ticket purchases. "No results found"  is shown instead

As well as the above, it is also possible to click on the "Turn on" button without setting a VSP or a purchase account.

Screenshot from 2020-11-03 12-13-35

### LINKED ISSUE #2856: [1.6.0-rc2] VSP is empty on auto buyer modal
The address of the VSP isn't being filled.

### PR DESCRIPTION
**Pull request by bgptr**
Closes #2852 
Closes #2856
Also, I noticed that although the host was set up correctly, the legacy auto buyer did not start (and could not be stopped). It was because a part of the relevant v2 code has been deleted at some point. I restored it and wired up to the legacy auto buyer page.

### CODE REVIEW

**Review by vctt94 — COMMENTED** [review]
Thanks for this PR. It is looking good. 

I noticed if you keep the `balance to mantain` field as 0, it won't start, although, 0 is a valid value.

Also, can you add an error if a field is missing, like we do with the new vsp auto buyer?

**Comment by bgptr** [comment]
Thank you for your review. I've fixed your review comments.

File: app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/LEGACY_TicketAutoBuyer/index.js [hunk 1]
```diff
@@ -151,11 +151,7 @@ class TicketAutoBuyer extends React.Component {
   }
 
   getIsFormValid() {
-    return (
-      this.getAccount() &&
-      this.state.balanceToMaintain > 0 &&
-      this.getStakePool()
-    );
+    return this.getAccount() && this.getStakePool();
```
Comment (vctt94) [inline app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/LEGACY_TicketAutoBuyer/index.js:161]: maybe check if balance to mantain is `>=0`?

Comment (bgptr) [reply app/components/views/TicketsPage/PurchaseTab/LEGACY_PurchasePage/LEGACY_TicketAutoBuyer/index.js]: Sure. I've added it.

**Review by vctt94 — APPROVED** [review]
LGTM

**Pull request merged**